packages/machine-extractor/src/types.ts

```diff
@@ -38,4 +78,64 @@ export interface FileExtractResult<
   file: t.File;
 }
 
-export type DeclarationType = 'named' | 'inline' | 'identifier' | 'unknown';
+export type DeclarationType =
+  | 'named'
+  | 'inline'
+  | 'identifier'
+  | 'object'
+  | 'unknown';
+
+type BaseBuiltinAction<P extends object> = {
+  kind: 'builtin';
+  action: { type: string } & P;
+};
+export type ExtractorAssignAction = BaseBuiltinAction<{
+  assignment:
+    | Record<string, JsonItem | JsonExpressionString>
+    | JsonExpressionString;
+}>;
+export type ExtractorRaiseAction = BaseBuiltinAction<{
+  event: Record<string, JsonItem> | JsonExpressionString;
+}>;
+export type ExtractedLogAction = BaseBuiltinAction<{
+  expr: string | JsonExpressionString;
+}>;
+export type ExtractorSendToAction = BaseBuiltinAction<{
+  event: JsonExpressionString | Record<string, JsonItem>;
+  to: string | JsonExpressionString;
+  id?: string | JsonExpressionString;
+  delay?: string | number | JsonExpressionString;
+}>;
+export type ExtractorStopAction = BaseBuiltinAction<{
+  id: string | JsonExpressionString;
+}>;
+
+export type NamedAction = {
+  kind: 'named';
+  action: { type: string; params?: JsonObject };
+};
+export type InlineAction = {
+  kind: 'inline';
+  action: { expr: JsonExpressionString };
+};
+export type BuiltinAction =
+  | ExtractorAssignAction
+  | ExtractorRaiseAction
+  | ExtractedLogAction
+  | ExtractorSendToAction
+  | ExtractorStopAction;
+
+export type MachineAction = NamedAction | InlineAction | BuiltinAction;
+
+// These types are copied over from studio blocks.
+// array and object are extracted as expressions so Studio render them correctly when exporting
+export type JsonItem =
+  | string
+  | number
+  | bigint
```

Comment:
bigint is not a valid JSON value - so this type should be somehow renamed, maybe `TransferrableItem`?